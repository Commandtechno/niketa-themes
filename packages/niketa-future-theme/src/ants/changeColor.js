import * as chromatism from 'chromatism'
import { switcher } from 'rambdax'

export function changeColorAnt(color, modeInput){
  const change = switcher(modeInput.toUpperCase())
    .is('DARKEST', -11)
    .is('DARKER', -8)
    .is('DARK', -5)
    .is('LIGHTEST', 11)
    .is('LIGHTER', 8)
    .is('LIGHT', 5)
    .default(0)

  const hasOpacity = color.length === 9
  if (!change || hasOpacity) return color

  return chromatism.brightness(change, color).hex
}
