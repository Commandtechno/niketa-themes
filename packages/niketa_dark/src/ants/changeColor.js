import Color from 'color'
import { switcher } from 'rambdax'

function whenNoOpacity({ color, change, mode }){
  return Color(color)[ mode ](change)
    .hex()
}

const BASE = 0.08

export function changeColorAnt(
  color, modeInput, base = BASE
){
  const { mode, change } = switcher(modeInput)
    .is('DARKEST', {
      mode   : 'darken',
      change : base * 3,
    })
    .is('DARKER', {
      mode   : 'darken',
      change : base * 2,
    })
    .is('DARK', {
      mode   : 'darken',
      change : base,
    })
    .is('LIGHTEST', {
      mode   : 'lighten',
      change : base * 3,
    })
    .is('LIGHTER', {
      mode   : 'lighten',
      change : base * 2,
    })
    .is('LIGHT', {
      mode             : 'lighten',
      change           : base,
      opacityChange    : 30,
      opacityDirection : 'minus',
    })
    .default({})

  if (!mode) return color

  return whenNoOpacity({
    color,
    change,
    mode,
  })
}
