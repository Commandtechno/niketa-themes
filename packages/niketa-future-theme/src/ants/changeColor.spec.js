import { mapToObject } from 'rambdax'

import { changeColorAnt } from './changeColor'
import { writeJsonAnt } from './writeJson'

test('happy', () => {
  const color = '#f26153'

  const modes = [
    'light',
    'lighter',
    'lightest',
    'origin',
    'dark',
    'darker',
    'darkest',
  ]
  const results = mapToObject(mode => ({ [ mode ] : changeColorAnt(color, mode) }))(modes)

  writeJsonAnt('src/ants/changeColor.json', results)
})
