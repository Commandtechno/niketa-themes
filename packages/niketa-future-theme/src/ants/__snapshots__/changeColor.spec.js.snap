// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "dark": "#f04b3b",
  "darker": "#ef3e2d",
  "darkest": "#ee311f",
  "light": "#f4776b",
  "lighter": "#f58479",
  "lightest": "#f69187",
  "origin": "#f26153",
}
`;
